citadel_workspace_service/src/kernel/request_handler.rs

```diff
@@ -1408,7 +1409,8 @@ pub async fn handle_request(
             let mut server_connection_map = server_connection_map.lock().await;
             if let Some(connection) = server_connection_map.get_mut(&cid) {
                 if let Some(group_connection) = connection.groups.get_mut(&group_key) {
-                    match group_connection.tx.leave().await {
+                    let group_sender = group_connection.tx.clone();
+                    match group_sender.leave().await {
```

Comment:
Right after cloning the group sender, make sure to `drop(server_connection_map)`, that way, we don't block the map while the `leave()` function gets called.

1. Clone
2. Drop the lock
3. Call the async function

Make sure to do this everywhere else that the previous review mentioned